lib/datasource/rubygems/get.js

```diff
@@ -0,0 +1,115 @@
+const URL = require('url');
+const got = require('got');
+const { satisfies } = require('@snyk/ruby-semver');
+const { maskToken } = require('../../util/mask');
+const hostRules = require('../../util/host-rules');
+const retriable = require('./retriable');
+const { wrap, httpCache } = require('./cacheable');
+const { UNAUTHORIZED, FORBIDDEN, NOT_FOUND } = require('./errors');
+
+const INFO_PATH = '/api/v1/gems';
+const VERSIONS_PATH = '/api/v1/versions';
+
+// istanbul ignore next
+const processError = ({ err, ...rest }) => {
+  const { statusCode, headers = {} } = err;
+  const data = {
+    ...rest,
+    err,
+    statusCode,
+    token: maskToken(headers.authorization) || 'none',
+  };
+
+  switch (statusCode) {
+    case NOT_FOUND:
+      logger.info(data, 'RubyGems lookup failure: not found');
+      break;
+    case FORBIDDEN:
+    case UNAUTHORIZED:
+      logger.info(data, 'RubyGems lookup failure: authentication failed');
+      break;
+    default:
+      logger.debug(data, 'RubyGems lookup failure');
+      throw new Error('registry-failure');
+  }
+};
+
+const getHeaders = url => {
+  const { host } = URL.parse(url.toString());
+  const { token } = hostRules.find({ host, platform: 'rubygems' }) || {};
+
+  return token ? { Authorization: token } : {};
+};
+
+const isCompatible = ({ ruby }, { rubyVersion }) =>
+  !ruby ? true : satisfies(ruby, rubyVersion);
+
+const fetch = async ({ dependency, registry, path }) => {
+  const json = true;
+
+  const retry = retriable();
+  const headers = getHeaders(registry);
+  const cache = process.env.RENOVATE_SKIP_CACHE ? undefined : httpCache;
+
+  const name = `/${dependency}.json`;
+  const baseUrl = `${registry}/${path}`;
+
+  const response =
+    (await got(name, { cache, retry, json, baseUrl, headers })) || {};
+
+  return response.body;
+};
+
+const getDependency = async ({ dependency, registry, compatibility }) => {
+  try {
+    const info = await fetch({ dependency, registry, path: INFO_PATH });
+    if (!info) {
+      logger.debug({ dependency }, 'RubyGems package not found.');
+      return null;
+    }
+
+    if (dependency.toLowerCase() !== info.name.toLowerCase()) {
+      logger.warn(
+        { lookup: dependency, returned: info.name },
+        'Lookup name does not match with returned.'
+      );
+      return null;
+    }
+
+    const versions =
+      (await fetch({ dependency, registry, path: VERSIONS_PATH })) || [];
+
+    const releases = versions
+      .map(
+        ({
+          platform,
+          prerelease,
+          number: version,
+          created_at: releaseDate,
+          rubygems_version: rubygemsVersion,
+          ruby_version: rubyVersion,
+        }) => ({
+          platform,
```

Comment:
Should we use platform anywhere?

Reply:
Hm, not sure about this. I add it cause you mentioned it [here](https://github.com/renovatebot/renovate/issues/2981))

Reply:
Ha, I forgot. Let's rename it as `rubyPlatform` but it's probably unused.

Reply:
Updated